lib/ClangImporter/ImportType.cpp

```diff
@@ -1986,9 +1979,11 @@ Type ClangImporter::Implementation::importMethodType(
         swiftResultTy = OptionalType::get(OptionalityOfReturn, swiftResultTy);
     }
   }
+
   if (!swiftResultTy)
     return Type();
-  swiftResultTy = mapTypeIntoContext(swiftResultTy);
+
+  swiftResultTy = mapTypeIntoContext(origDC, dc, swiftResultTy);
```

Comment:
But we shouldn't have contextual types (archetypes) at this point anyway, right?